### DISCUSSION TITLE
Why the docs recommend react-query over alternatives

### DISCUSSION THREAD

**Discussion by IGassmann** [dotansimha/graphql-code-generator#9482] [Q&A] [answered]
In the [React Vue Guide](https://the-guild.dev/graphql/codegen/docs/guides/react-vue), GraphQL Code Generator recommends using `@tanstack/react-query`.

What are the reasons behind recommending `react-query` over a more GraphQL-centric client such as URQL or Apollo Client? Are there some key benefits that `react-query` has over those alternatives? I'd love to hear your thoughts behind this recommendation. :)

**Comment by charpeni** [comment, marked as answer]
If you're referring to the following section:

<img width="500" alt="Screenshot 2023-06-05 at 4 04 37 PM" src="https://github.com/dotansimha/graphql-code-generator/assets/7189823/eb35a6b9-310c-430a-968e-e2bbd487f55e">

I believe it is meant to be an information block for `@tanstack/react-query` users, meaning that if you use `@tanstack/react-query`, it is recommended to use it with `graphql-request@^5.0.0` due to GraphQL Code Generator integration. Not that it is recommended to use GraphQL Code Generator with `@tanstack/react-query`. Most examples default to React Apollo and GraphQL Code Generator is working well with [most modern GraphQL clients and frameworks](https://the-guild.dev/graphql/codegen/docs/guides/react-vue#appendix-ii-compatibility).

Additionally, there's [an appendix for using React Query with a custom fetcher](https://the-guild.dev/graphql/codegen/docs/guides/react-vue#appendix-i-react-query-with-a-custom-fetcher-setup) in case you don't want to use the highly recommended one, `graphql-request@^5.0.0`.

**Reply by IGassmann** [reply]
It looks like I misinterpreted this section.

It might be worth rephrasing it to something similar to the following to avoid confusion for other readers :)

> If you're using `@tanstack/react-query`, we recommend using it with `graphql-request@^5.0.0`  to get the best developer experience.

**Reply by n1ru4l** [reply]
@IGassmann feel free to send a PR for updating the documentation!

**Reply by IGassmann** [reply]
Done https://github.com/dotansimha/graphql-code-generator/pull/9488 :)
